### PR TITLE
fix: skipRounding implemented for TE [DHIS2-17105]

### PR DESCRIPTION
**Pull request by gnespolino**
This PR adds support for `skipRounding` parameter in TE analytics endpoint

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/TeiListGrid.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  private ValueType getValueType(String col) {
+    return teiQueryParams
+        .getCommonParams()
+        .streamDimensions()
+        .filter(d -> d.toString().equals(col))
+        .findFirst()
+        .map(DimensionIdentifier::getDimension)
+        .map(DimensionParam::getValueType)
+        .orElse(ValueType.TEXT);
+  }
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/TeiListGrid.java:114]: This code was moved mostly unchanged from `ListGrid`

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/TeiListGrid.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        rowContext.put(getCurrentRowWriteIndex(), rowContextItems);
+      }
+    }
+
+    // Needs to ensure the ordering of columns based on grid headers.
+    repositionColumns(repositionHeaders(new ArrayList<>(headersSet)));
+
+    setRowContext(rowContext);
+    return this;
+  }
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/TeiListGrid.java:90]: this code was moved mostly unchanged from `ListGrid` and it is TE specific

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/TeiListGrid.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+    while (rs.next()) {
+      addRow();
+      Map<String, Object> rowContextItems = new HashMap<>();
+
+      for (int i = 0; i < cols.length; i++) {
+        if (headerExists(cols[i])) {
+          String columnLabel = cols[i];
+
+          Object value = getValueAndRoundIfNecessary(rs, columnLabel);
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/TeiListGrid.java:73]: here is where we decide if rounding the value or not

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/TeiListGrid.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+      if (valueStatus != ValueStatus.SET) {
+        Map<String, String> valueStatusMap = new HashMap<>();
+        valueStatusMap.put("valueStatus", valueStatus.getValue());
+        rowContextItem.put(Integer.toString(rowIndex), valueStatusMap);
+      }
+    }
+
+    return rowContextItem;
+  }
+}
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/common/TeiListGrid.java:161]: this code was moved mostly unchanged from `ListGrid` and it is TE specific

**Review by maikelarabori — APPROVED** [review]

**Review by d-bernat — APPROVED** [review]

**Pull request merged**